### PR TITLE
ROE-2357 Update readme for node 18 and ECS

### PR DESCRIPTION
**Pull request by mattch1**
### JIRA link
https://companieshouse.atlassian.net/browse/ROE-2357

### Change description

Updated readme by 
- Specifying node 18 rather than node 16.
- Introducing terraform.
- Adding the variable required by terraform.

Corrected a spelling mistake in the terraform variables.

### Work checklist

- [ ] Tests added where applicable NA
- [ ] UI changes meet accessibility criteria NA

### Merge instructions

We are committed to keeping commit history clean, consistent and linear. To achieve this, this commit should be structured as follows:

```
<type>[optional scope]: <description>
```

and contain the following structural elements:

- fix: a commit that patches a bug in your codebase (this correlates with PATCH in semantic versioning),
- feat: a commit that introduces a new feature to the codebase (this correlates with MINOR in semantic versioning),
- BREAKING CHANGE: a commit that has a footer `BREAKING CHANGE:` introduces a breaking API change (correlating with MAJOR in semantic versioning). A BREAKING CHANGE can be part of commits of any type,
- types other than `fix:` and `feat:` are allowed, for example `build:`, `chore:`, `ci:`, `docs:`, `style:`, `refactor:`, `perf:`, `test:`, and others,
- footers other than `BREAKING CHANGE: <description>` may be provided.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -92,6 +92,62 @@ PIWIK_UPDATE_START_GOAL_ID | Matomo Update Start goal ID | 3
 SHOW_SERVICE_OFFLINE_PAGE | Feature Flag | false
 VF01_FORM_DOWNLOAD_URL | Overseas entity verification checks statement URL | `https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/1095139/OE_VF01.pdf`
 
+
+## Terraform and variables required
+
+Terraform is run via a concourse ci pipeline is used in place of AWS Cloud Formation to deploy the cluster and services.
```
Comment (markpit) [inline README.md:98]: not sure if there is a word missing in this sentence, maybe an 'and', it didn't quite read right to me - Terraform is run via a concourse ci pipeline (and?) is used in place of AWS Cloud Formation

Comment (mattch1) [reply README.md]: Some bad editing when merging previous sentences,  added "and" then updated the sentence.

File: README.md [hunk 2]
```diff
@@ -92,6 +92,62 @@ PIWIK_UPDATE_START_GOAL_ID | Matomo Update Start goal ID | 3
 SHOW_SERVICE_OFFLINE_PAGE | Feature Flag | false
 VF01_FORM_DOWNLOAD_URL | Overseas entity verification checks statement URL | `https://www.gov.uk/government/uploads/system/uploads/attachment_data/file/1095139/OE_VF01.pdf`
 
+
+## Terraform and variables required
+
+Terraform is run via a concourse ci pipeline and is used in place of AWS Cloud Formation for both provisioning and for deploying the cluster and services.
```
Comment (mwestacott) [inline README.md:98]: Reads well, but I'm just curious where the new Terraform details came from?

Comment (mattch1) [reply README.md:98]: It's a mix of what is in the link referenced, what we have done so far with terraform and from what I have learned about AWS, Cloud Formation is AWS's own IoC tool for deployment.

**Review by markpit — APPROVED** [review]

**Review by dlloyd1-ch — APPROVED** [review]

**Review by mwestacott — COMMENTED** [review]
Might have spotted another required change - line 57 that has the command to manually build the Docker image...

Should it be referencing `416670754337` rather than `169942020521` and `eu-west-2` rather than `eu-west-1` now?

Worth adding details here anyway, to specify the AWS regions that you need to be logged into, I think.

I would try the Docker Build command locally @mattch1 to see if it does need changing in the README.

**Review by mwestacott — APPROVED** [review]

File: README.md [hunk 3]
```diff
@@ -54,7 +54,18 @@ Environment variables used to configure this service in docker are located in th
 
 ### To build the Docker container
 
-1. `DOCKER_BUILDKIT=0 docker build --build-arg SSH_PRIVATE_KEY="$(cat ~/.ssh/id_rsa)" --build-arg SSH_PRIVATE_KEY_PASSPHRASE -t 169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/overseas-entities-web .`
+As of the node 18 upgrade, the app is NO longer built in the eu-west-1 region with:
+
+`DOCKER_BUILDKIT=0 docker build --build-arg SSH_PRIVATE_KEY="$(cat ~/.ssh/id_rsa)" --build-arg SSH_PRIVATE_KEY_PASSPHRASE -t 169942020521.dkr.ecr.eu-west-1.amazonaws.com/local/overseas-entities-web .`
+
```
Comment (dlloyd1-ch) [inline README.md:60]: These line are superfluous and will become dated quite quickly. Suggest removing them and the word 'INSTEAD' on line 61

Comment (mattch1) [reply README.md]: Removed old command.

**Review by dlloyd1-ch — APPROVED** [review]

**Pull request merged**